### PR TITLE
Use the random library

### PR DESCRIPTION
**Pull request by magoorden**
It is recommended to no longer use `rand()` for random number generation.

### CODE REVIEW

File: test/test_cdd.cpp [hunk 1]
```diff
@@ -32,10 +33,26 @@ using std::cerr;
 using std::cout;
 using base::Timer;
 
-/** Random range, may change definition */
-#define RANGE()      ((rand() % 10000) + 1)
-#define RANDOMINT(n) ((rand() % n))
-#define RANDOMBOOL() (rand() > (RAND_MAX / 2))
+/** Random helper functions. */
```
Comment (RasmusRendal) [inline test/test_cdd.cpp:36]: Maybe it's not a good idea to reinitialize the random number generator every time it's called? We could also just use the random number generation included in UUtils: https://github.com/UPPAALModelChecker/UUtils/blob/main/include/base/random.h

Comment (magoorden) [reply test/test_cdd.cpp:36]: Thanks for the reference to UUtils. Let me see what I can do with that.

**Review by RasmusRendal — APPROVED** [review]

**Comment by magoorden** [comment]
I'm not a C/C++ expert, but I was thinking about the following. Now I create a new `RandomGenerator` object each time a random number is requested (using the static seed). Therefore, no `RandomGenerator` object (pointer) has to be passed around to all the tests. It simplifies the code, but creating these objects all the time might have (possible) negative consequences for the performec if we have a lot of tests to run. What do you think?

**Comment by RasmusRendal** [comment]
> I'm not a C/C++ expert, but I was thinking about the following. Now I create a new `RandomGenerator` object each time a random number is requested (using the static seed). Therefore, no `RandomGenerator` object (pointer) has to be passed around to all the tests. It simplifies the code, but creating these objects all the time might have (possible) negative consequences for the performce if we have a lot of tests to run. What do you think?

I don't think it's worth thinking about performance yet, since it's just in the test, and the test takes less than a fifth of a second as of now. If we ever have the test take longer than that, we can have a random generator with a static lifetime.

**Pull request merged**